java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -939,6 +939,9 @@ private void handleSubscribeChannels(ServerAction serverAction, JsonElement json
                     newTokenIds);
         }
         else {
+            //set the token as invalid
+            SubscribeChannelsAction sca = (SubscribeChannelsAction)action;
+            sca.getDetails().getAccessTokens().forEach(x -> x.setValid(false));
```

Comment:
Have you tested that this really change the value in the DB?

Reply:
no :) I'm doing it now

Reply:
ok, the current version saves the valid token variable correctly. Just to confirm:
- I checked the variable in susechannelaccesstoken
- I simulate a failure on subscribe channel action making file `/etc/zypp/repos.d/susemanager:channels.repo` immutable